airflow/jobs.py

```diff
@@ -622,6 +622,7 @@ def manage_slas(self, dag, session=None):
             self.log.info("Skipping SLA check for %s because no tasks in DAG have SLAs", dag)
             return
 
+        # Find the last executed tasks
```

Comment:
If I'm not mistaken, this is not find the last executed tasks but rather, find the last succesful or skipped task instances.
Which by the way is a bit weird to base SLAs from only successful tasks instances. What if the the whole history has failed ? We don't get any SLA ? I think we need to change the filter down below too so it uses any task that hasn't be scheduled yet as a basis.

Reply:
Nevermind, #3584 adresses the problem perfectly.

Reply:
Exactly, I mention on the ticket there are couple of open issues regarding this topic and this is my first cleanup attempt, without affecting functionality.